### PR TITLE
Fix indented code blocks in docstrings

### PR DESCRIPTION
**Pull request by gracepetryk**
This PR fixes an issue where markdown code blocks do not parse correctly when they started on an indented line in a docstring.

```python
def test():
    """
    A test function

    Example:
        ```python
        print(test())
        ```
    """
    pass
```

Currently this docstring looks like this when hovering `test`: 
![image](https://user-images.githubusercontent.com/5047809/222246442-6db923e7-8d59-4236-84e0-d42ada48295d.png)

After the fix it renders correctly:
![image](https://user-images.githubusercontent.com/5047809/222276577-57ad01d1-5be2-41b3-92bd-4e5a71bdd8ef.png)

### CODE REVIEW

File: packages/pyright-internal/src/analyzer/docStringConversion.ts [hunk 1]
```diff
@@ -431,7 +436,8 @@ class DocStringConverter {
     }
 
     private _parseBacktickBlock(): void {
-        if (this._currentLine().startsWith('```')) {
+        // only match closing ``` at same indent level of opening
```
Comment (erictraut) [inline packages/pyright-internal/src/analyzer/docStringConversion.ts:439]: Nit: Comments within the pyright code base begin with capital letters and end in a period if they are complete sentences.

File: packages/pyright-internal/src/analyzer/docStringConversion.ts [hunk 2]
```diff
@@ -421,8 +424,10 @@ class DocStringConverter {
     }
 
     private _beginBacktickBlock(): boolean {
-        if (this._currentLine().startsWith('```')) {
-            this._appendLine(this._currentLine());
+        const match = this._currentLine().match(CodeBlockStartRegExp);
+        if (match !== null) {
+            this._blockIndent = this._currentIndent()
+            this._appendLine('```' + match[1]) // remove indentation, preserve language tag
```
Comment (erictraut) [inline packages/pyright-internal/src/analyzer/docStringConversion.ts:430]: Nit: Don't use end-of-line comments. Move to previous line. Comments should begine with capital letter and end in period if they are complete sentence.

**Review by erictraut — CHANGES_REQUESTED** [review]
There should be a bug report associated with the change.

Change should also include a test to valid the change and prevent regressions.

**Comment by gracepetryk** [comment]
https://github.com/microsoft/pyright/issues/4718

**Comment by gracepetryk** [comment]
@microsoft-github-policy-service agree

File: packages/pyright-internal/src/analyzer/docStringConversion.ts [hunk 3]
```diff
@@ -421,8 +424,13 @@ class DocStringConverter {
...
 
     private _beginBacktickBlock(): boolean {
-        if (this._currentLine().startsWith('```')) {
-            this._appendLine(this._currentLine());
+        const match = this._currentLine().match(CodeBlockStartRegExp);
+        if (match !== null) {
+            this._blockIndent = this._currentIndent()
+
+            // Remove indentation and preserve language tag.
+            this._appendLine('```' + match[1])
```
Comment (debonte) [inline packages/pyright-internal/src/analyzer/docStringConversion.ts:432]: Match group would be null if the language tag is missing?

**Review by debonte — CHANGES_REQUESTED** [review]

File: packages/pyright-internal/src/analyzer/docStringConversion.ts (hunk 3 above)
Comment (gracepetryk) [reply packages/pyright-internal/src/analyzer/docStringConversion.ts:432]: match returns an empty string if a capture group is empty
```
> '    ```'.match(/\s*```(\w*)/)[1]
''
```

**Review by erictraut — APPROVED** [review]
Looks good to me. Thanks for the contribution!

**Pull request merged**